src/screenshotdialog.c

```diff
@@ -97,11 +422,12 @@ screenshot_dialog_new (const char *app_id,
 
   gtk_label_set_label (GTK_LABEL (dialog->heading), heading);
 
-  pixbuf = gdk_pixbuf_new_from_file_at_scale (filename, 500, 400, TRUE, &error);
-  if (error)
-    g_warning ("Failed to load screenshot %s: %s", filename, error->message);
+  dialog->shell = shell;
```

Comment:
Looks like `shell` is declared as an static global variable in screenshot.c so it's probably not needed, but I still think increasing the ref count here and decreasing it in finalize() would probably be a good idea.

Reply:
fixed